tests/lib/rules/no-unassigned-vars.js

```diff
@@ -0,0 +1,106 @@
+/**
+ * @fileoverview Tests for no-unassigned-vars rule.
+ * @author Jacob Bandes-Storch <https://github.com/jtbandes>
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/no-unassigned-vars"),
+	RuleTester = require("../../../lib/rule-tester/rule-tester");
+const { unIndent } = require("../../_utils");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+	languageOptions: {
+		ecmaVersion: 2022,
+		sourceType: "script",
+	},
+});
+
+ruleTester.run("no-unassigned-vars", rule, {
+	valid: [
+		"let x;",
+		"var x;",
+		"const x = undefined; log(x);",
+		"let y = undefined; log(y);",
+		"var y = undefined; log(y);",
+		"let a = x, b = y; log(a, b);",
+		"var a = x, b = y; log(a, b);",
+		"const foo = (two) => { let one; if (one !== two) one = two; }",
+	],
+	invalid: [
+		{
+			code: "let x; let a = x, b; log(x, a, b);",
+			errors: [
+				{ messageId: "unassigned", line: 1, column: 5 },
```

Comment:
Can you also add `data` property to all errors in invalid test cases?

https://eslint.org/docs/latest/integrate/nodejs-api#testing-errors-with-messageid